File: Data Structures & Algorithms/two-integer-sum/submission-1.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = hashMapOf<Int, Int>();
        for(i in nums.indices){
            val complement = target - nums[i];
            if(map.containsKey(complement)){
                return intArrayOf(map[complement]!!, i)
            }else{
                map[nums[i]] = i
            }
            print("Hashmap : $map")
        }
        return intArrayOf()
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        if(nums.size() == 1){
            return nums[0];
        }
        for(int i=0; i < nums.size() - 1; i++){
            int left = nums[i];
            int right = nums[i + 1];
            if(left > right){
                return right;
            }
        }
        return nums[0];
    }
};
